### PR TITLE
Add pragma for documentation-only, move isSubtype etc to Types

### PR DESCRIPTION
**Pull request by mppf**
* Adds a flag to the compiler, `pragma "docs only"` / `FLAG_DOCS_ONLY`
* Adjusts the `cleanup` pass to remove functions marked with 
  `FLAG_DOCS_ONLY`. These will still be visited by `chpldoc` because it
  generates docs before `cleanup`.
* Move docs for `isSubtype` etc from `UtilMisc_forDocs` to `Types`
* Remove `UtilMisc_forDocs` since it is now empty.

Reviewed by @lydia-duncan - thanks!

- [x] full local testing

### CODE REVIEW

File: compiler/include/flags_list.h [hunk 1]
```diff
@@ -119,6 +119,7 @@ symbolFlag( FLAG_DEFAULT_INTENT_IS_REF_MAYBE_CONST, ypr, "default intent is ref
 symbolFlag( FLAG_COPY_INIT, npr, "copy initializer", ncm )
 symbolFlag( FLAG_DESTRUCTOR , npr, "destructor" , "applied to functions that are destructors" )
 symbolFlag( FLAG_DISTRIBUTION , ypr, "distribution" , ncm )
+symbolFlag( FLAG_DOCS_ONLY, ypr, "docs only", "ignore this symbol when compiling" )
```
Comment (lydia-duncan) [inline compiler/include/flags_list.h:122]: I'd say "intended for documentation purposes only" or something like that

File: doc/rst/language/spec/types.rst [hunk 2]
```diff
@@ -660,17 +660,17 @@ function (see :ref:`Formal_Type_Arguments`).
...
 The normal comparison operators are also available to compare types:
 
  * ``==`` checks if two types are equivalent
  * ``!=`` checks if two types are different
  * ``<`` and ``>`` check if one type is a proper subtype of another (see
-   :proc:`< <UtilMisc_forDocs.<>`)
+   :proc:`< <Types.<>`)
  * ``<=`` and ``>=`` check if one type is a subtype of another (see
-   :proc:`<= <UtilMisc_forDocs.<=>`)
+   :proc:`<= <Types.<=>`)
```
Comment (lydia-duncan) [inline doc/rst/language/spec/types.rst:673]: Is it possible to add some buffer space here? E.g. `< Types.<= >`.  I find it a little difficult to read, but I'm not certain if it's required

**Review by lydia-duncan — COMMENTED** [review]
Just a couple of comments, thanks!

File: doc/rst/language/spec/types.rst (hunk 2 above)
Comment (mppf) [reply doc/rst/language/spec/types.rst:673]: Unfortunately, it does not work if I do that.

**Review by lydia-duncan — APPROVED** [review]

**Pull request merged**